astrokat/noisediode.py

```diff
@@ -9,257 +9,408 @@
     from katcorelib import user_logger
 except ImportError:
     from .simulate import user_logger
+from . import _DEFAULT_LEAD_TIME
+from . import max_cycle_len
 
 
-_DEFAULT_LEAD_TIME = 5.0  # lead time [sec]
+def _get_max_cycle_len(kat):
+    """Get maximum cycle length for noise diode switching
+    """
+    if not kat.dry_run:
+        return max_cycle_len(kat.sensor.sub_band.get_value())
```

Comment:
Any likeliness of this causing an `AttributeError`, perhaps check if it has the attr first (`hasattr`) - else ignore comment.

Reply:
@mmphego my skill set is a little lacking to quickly answer your question :(

Am I understanding correctly that you suggest I check that a `sub_band` sensor is available before trying to get the value, i.e.

```
if hasattr(kat.sensor, 'sub_band'):
    band_max_cycle_len = max_cycle_len(kat.sensor.sub_band.get_value())
return band_max_cycle_len
```

Reply:
It is not likely.  The subarray must have a "band" sensor, and the kat container object must have access to it.  If the attribute is missing, then something critical is broken in the kat container object, and the observation should error out.